static/js/biomedical/disease/data_processing_utils.ts

```diff
@@ -232,3 +233,84 @@ export function getCompoundDiseaseTreatment(
   }
   return rawData;
 }
+
+/**
+ * Fetches the chemical compound disease contraindication data
+ * @param data
+ * @returns
+ */
+export function getCompoundDiseaseContraindication(
```

Comment:
The logic in this function is exactly the same as the last function `getCompoundDiseaseTreatment`, except for the check for the typeField (line 305) and the property (line 275), could we extract out a shared helper function that takes those two strings as arguments? I know the return type is also slightly different, but the shared function can return the raw value and then the individual functions can map that value to what is needed.

Reply:
That's a good point. I'm still trying to understand how do we return the raw data because the two return data types/interfaces would be different. And what part of the code would we want in the individual functions?

Reply:
The return type can be something like (w/ better naming)

```
DataType {
value: string;
id: string;
name: string;
node: string;
}
```

And each individual function would just call the shared function & do a map on the result to convert the value to the form needed.